src/assets/scripts/base/PositionModel.js

```diff
@@ -95,37 +161,38 @@ export default class PositionModel {
 
         this.latitude = parseCoordinate(coordinates[0]);
         this.longitude = parseCoordinate(coordinates[1]);
-        // GPS coordinates in [x,y] order
-        this.gps = [
-            this.longitude,
-            this.latitude
-        ];
 
         // TODO: this is using coersion and shoudld be updated to be more explicit
         if (coordinates[2] != null) {
             this.elevation = parseElevation(coordinates[2]);
         }
 
-      // this function (parse4326) is moved to be able to call it if point is
-      // EPSG:4326, numeric decimal, like those from GeoJSON
-        if (this.reference_position != null) {
-            this.x = this.longitude;
-            this.y = this.latitude;
-
-            this.parse4326();
+        // if there is no reference_position, x and y are both 0 and we don't have enough
+        // information to run `parse4326()`
+        if (!this.reference_position) {
+            return;
         }
+
+        // this function (parse4326) is moved to be able to call it if point is
+        // EPSG:4326, numeric decimal, like those from GeoJSON
+        // FIXME: why do x/y get assigned with lat/long here, and then in parse4326 lat/long gets assigned with x/y?
+        this.x = this.longitude;
```

Comment:
No clue whatsoever. This section has always confused me. All I can say is that `x` and `y` are used for positions within the canvas used for drawing, and `latitude` and `longitude` are used for everything else.